e2e-tests/e2e-tests-utils/src/test/java/org/evomaster/e2etests/utils/GraphqlTestBase.java

```diff
@@ -0,0 +1,221 @@
+
+            handleFlaky(
+                    () -> lambda.accept(new ArrayList<>(args))
+            );
+        });
+    }
+
+    protected void runTestHandlingFlakyAndCompilation(
+            String outputFolderName,
+            String fullClassName,
+            int iterations,
+            Consumer<List<String>> lambda) throws Throwable {
+
+        runTestHandlingFlaky(outputFolderName, fullClassName, iterations, lambda);
+
+        assertTimeoutPreemptively(Duration.ofMinutes(2), () -> {
+            ClassName className = new ClassName(fullClassName);
+            compileRunAndVerifyTests(outputFolderName, className);
+        });
+    }
+
+    protected void compileRunAndVerifyTests(String outputFolderName, ClassName className){
+
+        Class<?> klass = loadClass(className);
+        assertNull(klass);
+
+        compile(outputFolderName);
+        klass = loadClass(className);
+        assertNotNull(klass);
+
+        TestExecutionSummary summary = JUnitTestRunner.runTestsInClass(klass);
+        assertTrue(summary.getContainersFoundCount() > 0);
+        assertEquals(0, summary.getContainersFailedCount());
+        assertTrue(summary.getContainersSucceededCount() > 0);
+    }
+
+    protected void clearGeneratedFiles(String outputFolderName, ClassName testClassName){
+
+        File folder = new File(outputFolderPath(outputFolderName));
+        try{
+            FileUtils.deleteDirectory(folder);
+        }catch (Exception e){
+            throw new RuntimeException(e);
+        }
+
+        String bytecodePath = "target/test-classes/" + testClassName.getAsResourcePath();
+        File compiledFile = new File(bytecodePath);
+        compiledFile.delete();
+    }
+
+    protected Class<?> loadClass(ClassName className){
+        try {
+            return this.getClass().getClassLoader().loadClass(className.getFullNameWithDots());
+        } catch (ClassNotFoundException e) {
+            return null;
+        }
+    }
+
+    protected void compile(String outputFolderName){
+
+        CompilerForTestGenerated.INSTANCE.compile(
+                OutputFormat.KOTLIN_JUNIT_5,
+                new File(outputFolderPath(outputFolderName)),
+                new File("target/test-classes")
+        );
+    }
+
+    protected List<String> getArgsWithCompilation(int iterations, String outputFolderName, ClassName testClassName){
+
+        return new ArrayList<>(Arrays.asList(
+                "--createTests", "true",
+                "--seed", "42",
+                "--sutControllerPort", "" + controllerPort,
+                "--maxActionEvaluations", "" + iterations,
+                "--stoppingCriterion", "FITNESS_EVALUATIONS",
+                "--outputFolder", outputFolderPath(outputFolderName),
+                "--outputFormat", OutputFormat.KOTLIN_JUNIT_5.toString(),
+                "--testSuiteFileName", testClassName.getFullNameWithDots()
+        ));
```

Comment:
this could be overriden when extending RestTestBase, by adding something like  `--problemType=GRAPHQL`